server/server.go

```diff
@@ -315,6 +310,11 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	// Execute the optional request callback if it exists and it's not disabled
+	if s.ReqCallback != nil && !s.vhosts[host].config.TLS.Manual && s.ReqCallback(w, r) {
+		return
+	}
+
```

Comment:
The original code occurred before host was set.  Rather than duplicate code that was already there, I figured I should move it below the code that sets host and the code right after it that establishes fallback behavior in case host isn't in s.vhosts.

Moving after the setting of host was necessary for me to use that variable, but moving after the fallback code was a bit of a judgement call.  Since the fallback code can change the value of host and I believed that all vhost-specific behavior should be made to access the same record in a given call for the sake of consistency (If something came up later that my testing somehow missed or even resulted form an unrelated bug, having host pull some behavior from one vhosts entry (null behavior from having a host not in vhosts) and some behavior from another (the fallback entry) would make it unnecessarily difficult to see what's happening).